src/screens/TokenDetail.js

```diff
@@ -0,0 +1,685 @@
+   */
+  unregisterConfirmed = () => {
+    $('#confirmModal').modal('hide');
+    tokens.unregisterToken(this.state.token.uid);
+    this.props.history.push('/wallet/');
+  }
+
+  /**
+   * Called when user clicks to edit the token, then  opens the modal
+   */
+  editClicked = () => {
+    $('#editTokenModal').modal('show');
+  }
+
+  /**
+   * When user finish editing the token, closes the modal
+   */
+  editSuccess = (token) => {
+    $('#editTokenModal').modal('hide');
+    this.setState({ token });
+    this.showSuccess('Token edited!');
+  }
+
+  /**
+   * Called when user clicks to download the qrcode
+   * Add the href from the qrcode canvas
+   *
+   * @param {Object} e Event emitted by the link clicked
+   */
+  downloadQrCode = (e) => {
+    e.currentTarget.href = document.getElementsByTagName('canvas')[0].toDataURL();
+  }
+
+  /**
+   * Show alert success message
+   *
+   * @param {string} message Success message
+   */
+  showSuccess = (message) => {
+    this.setState({ successMessage: message }, () => {
+      this.refs.alertSuccess.show(3000);
+    })
+  }
+
+  /**
+   * Method called on copy to clipboard success  
+   * Show alert success message
+   *
+   * @param {string} text Text copied to clipboard
+   * @param {*} result Null in case of error
+   */
+  copied = (text, result) => {
+    if (result) {
+      // If copied with success
+      this.showSuccess('Configuration string copied to clipboard!');
+    }
+  }
+
+  /**
+   * Called when user clicks an action link
+   *
+   * @param {Object} e Event emitted by the link clicked
+   * @param {string} action String representing the action clicked
+   */
+  actionClicked = (e, action) => {
+    e.preventDefault();
+    this.cleanStates();
+    this.setState({ action });
+  }
+
+  /**
+   * Goes to initial state, without any action selected
+   */
+  cancelAction = () => {
+    this.cleanStates();
+  }
+
+  /**
+   * Clean all states to its initial values
+   */
+  cleanStates = () => {
+    this.setState({ action: '', errorMessage: '', pin: '', formValidated: false, loading: false });
+  }
+
+  /**
+   * Handle methods promise resolve and rejection
+   * Clean states, show messages of error/success
+   *
+   * @param {Promise} promise Promise returned from management method
+   * @param {string} successMessage Message to show in case of success
+   */
+  handlePromise = (promise, successMessage) => {
+    promise.then(() => {
+      this.cleanStates();
+      this.showSuccess(successMessage);
+    }, (message) => {
+      this.setState({ loading: false, errorMessage: message });
+    });
+  }
+
+  /**
+   * Execute destroy method
+   *
+   * @param {Object} array Array of outputs that we are destroying
+   * @param {string} label Label to show on the success message
+   */
+  executeDestroy = (array, label) => {
+    const data = [];
+    // Get the number of outputs the user requested to destroy in the expected format
```

Comment:
Isn't it better to let the user choose which one he would like to destroy? A possible scenario would be to have delegated the authority to another address in a sub-level of the HD Wallet. Then, he would like to revoke the authority of that wallet for any reason.

Reply:
I think that destroying a mint/melt output is an advanced feature and choosing which one to destroy is really advanced. To be honest I would go with the simple approach right now, so we keep the UI/UX simple and write this task to remember to be done in the future. I feel this is not a case that will be common now.

What do you think?

Reply:
I agree. We may go forward as it is. But I think this task may be common for token administrators. We can definitely do it in the future. Let's just create a card for it, please.

Reply:
Done